frontend/views/containers/contributions/IncomeDetails.vue

```diff
@@ -0,0 +1,248 @@
+              name='incomeDetailsType'
+              value='incomeAmount'
+              v-model='$v.form.incomeDetailsType.$model'
+              @change='resetAmount'
+            )
+            i18n No, I don't
+        transition(name='expand')
+          fieldset(v-if='!!form.incomeDetailsType')
+            label.field
+              .label {{ needsIncome ? L("What's your monthly income?") : L('How much do you want to pledge?') }}
+              .input-combo(
+                :class='{"error": $v.form.amount.$error }'
+                v-error:amount=''
+              )
+                input.input(
+                  type='number'
+                  v-model='$v.form.amount.$model'
+                )
+                .suffix {{ groupMincomeSymbolWithCode }}
+              .helper(v-if='needsIncome && whoIsPledging.length')
+                text-who(:who='whoIsPledging')
+                | &nbsp
+                i18n will ensure you meet the mincome
+              i18n.helper(v-else-if='!needsIncome') Define up to how much you pledge to contribute to the group each month. Only the minimum needed amount will be given.
+            payment-methods(selected='manual')
+        .buttons
+          i18n.is-outlined(tag='button' @click='closeModal') Cancel
+          i18n.is-success(tag='button' type='submit' :disabled='$v.form.$invalid') Save
+        // TODO/OPTIMIZE - create directive similar to vError
+        .c-feedback.has-text-1(
+          v-if='ephemeral.formFeedbackMsg.text'
+          :class='ephemeral.formFeedbackMsg.class'
+        ) {{ephemeral.formFeedbackMsg.text}}
+      .c-graph
+        | WIP: A graph is on its way!
+        //
+          group-pledges-graph.c-graph(
+            :userpledgeamount='!needsIncome && !!form.pledge ? Number(form.pledge) : null'
+            :userincomeamount='needsIncome && !!form.income ? Number(form.income) : null'
+          )
+</template>
+
+<script>
+import { mapGetters } from 'vuex'
+import { validationMixin } from 'vuelidate'
+import { required } from 'vuelidate/lib/validators'
+import sbp from '~/shared/sbp.js'
+import InputAmount from './InputAmount.vue'
+import PaymentMethods from './PaymentMethods.vue'
+import Tooltip from '@components/Tooltip.vue'
+import ModalBaseTemplate from '@components/Modal/ModalBaseTemplate.vue'
+import TextWho from '@components/TextWho.vue'
+import GroupPledgesGraph from '../GroupPledgesGraph.vue'
+import L from '@view-utils/translations.js'
+
+export default {
+  name: 'IncomeDetails',
+  mixins: [validationMixin],
+  components: {
+    ModalBaseTemplate,
+    InputAmount,
+    TextWho,
+    Tooltip,
+    PaymentMethods,
+    GroupPledgesGraph
+  },
+  data () {
+    return {
+      ephemeral: {
+        formFeedbackMsg: {}
+      },
+      form: {
+        incomeDetailsType: null,
+        amount: null
+      },
+      fakeStore: {
+        incomeDetailsType: 'pledgeAmount', // incomeAmount || pledgeAmount
+        amount: 100
+      }
```

Comment:
This `fakeStore` object should no longer exist

Reply:
I think this is out of scope. We will remove this together on next PR

Reply:
It's in scope. It is already done in Contributions.vue on `master`.